Objects/abstract.c

```diff
@@ -2812,7 +2812,7 @@ object_issubclass(PyThreadState *tstate, PyObject *derived, PyObject *cls)
         return -1;
     }
 
-    /* Probably never reached anymore. */
+    /* Can be reached when infinite recursion happens. */
```

Comment:
How this change is related to this PR. Are there any tests for this?

Reply:
Yes, it is tested in `test_forwardref_subclass_type_error`.

This was changed, because when I was reading through `issubclass` source, I found this statement, which is wrong.

Proof: change `return recursive_issubclass(derived, cls);` to `Py_UNREACHABLE();`

And run tests:

```
» ./python.exe -m test -v test_typing
== CPython 3.13.0a0 (heads/issue-101162-dirty:755a0c6dacd, Aug 10 2023, 12:44:42) [Clang 14.0.3 (clang-1403.0.22.14.1)]
== macOS-13.4.1-arm64-arm-64bit little-endian
== Python build: debug
== cwd: /Users/sobolev/Desktop/cpython/build/test_python_68428æ
== CPU count: 12
== encodings: locale=UTF-8, FS=utf-8
0:00:00 load avg: 1.15 Run tests sequentially
0:00:00 load avg: 1.15 [1/1] test_typing

...

test_forwardref_subclass_type_error (test.test_typing.ForwardRefTests.test_forwardref_subclass_type_error) ... Fatal Python error: object_issubclass: We've reached an unreachable state. Anything is possible.
The limits were in our heads all along. Follow your dreams.
https://xkcd.com/2200
Python runtime state: initialized

Current thread 0x00000001eb689e00 (most recent call first):
  File "/Users/sobolev/Desktop/cpython/Lib/test/test_typing.py", line 5436 in test_forwardref_subclass_type_error
  File "/Users/sobolev/Desktop/cpython/Lib/unittest/case.py", line 589 in _callTestMethod
  File "/Users/sobolev/Desktop/cpython/Lib/unittest/case.py", line 634 in run
  File "/Users/sobolev/Desktop/cpython/Lib/unittest/case.py", line 690 in __call__
  File "/Users/sobolev/Desktop/cpython/Lib/unittest/suite.py", line 122 in run
  File "/Users/sobolev/Desktop/cpython/Lib/unittest/suite.py", line 84 in __call__
  File "/Users/sobolev/Desktop/cpython/Lib/unittest/suite.py", line 122 in run
  File "/Users/sobolev/Desktop/cpython/Lib/unittest/suite.py", line 84 in __call__
  File "/Users/sobolev/Desktop/cpython/Lib/unittest/suite.py", line 122 in run
  File "/Users/sobolev/Desktop/cpython/Lib/unittest/suite.py", line 84 in __call__
  File "/Users/sobolev/Desktop/cpython/Lib/unittest/runner.py", line 240 in run
  File "/Users/sobolev/Desktop/cpython/Lib/test/support/__init__.py", line 1116 in _run_suite
  File "/Users/sobolev/Desktop/cpython/Lib/test/support/__init__.py", line 1242 in run_unittest
  File "/Users/sobolev/Desktop/cpython/Lib/test/libregrtest/runtest.py", line 294 in _test_module
  File "/Users/sobolev/Desktop/cpython/Lib/test/libregrtest/runtest.py", line 330 in _runtest_inner2
  File "/Users/sobolev/Desktop/cpython/Lib/test/libregrtest/runtest.py", line 373 in _runtest_inner
  File "/Users/sobolev/Desktop/cpython/Lib/test/libregrtest/runtest.py", line 248 in _runtest
  File "/Users/sobolev/Desktop/cpython/Lib/test/libregrtest/runtest.py", line 278 in runtest
  File "/Users/sobolev/Desktop/cpython/Lib/test/libregrtest/main.py", line 483 in run_tests_sequential
  File "/Users/sobolev/Desktop/cpython/Lib/test/libregrtest/main.py", line 621 in run_tests
  File "/Users/sobolev/Desktop/cpython/Lib/test/libregrtest/main.py", line 799 in _main
  File "/Users/sobolev/Desktop/cpython/Lib/test/libregrtest/main.py", line 758 in main
  File "/Users/sobolev/Desktop/cpython/Lib/test/libregrtest/main.py", line 822 in main
  File "/Users/sobolev/Desktop/cpython/Lib/test/__main__.py", line 2 in <module>
  File "/Users/sobolev/Desktop/cpython/Lib/runpy.py", line 88 in _run_code
  File "/Users/sobolev/Desktop/cpython/Lib/runpy.py", line 198 in _run_module_as_main

Extension modules: _testcapi (total: 1)
[1]    68428 abort      ./python.exe -m test -v test_typing
                                                              
```